### ISSUE TITLE
viewing projects from external links takes 5-6s to load

### ISSUE THREAD

**Issue by pianoman19372** [AWeber-Imbi/imbi#53]
i'm linking external documentation to projects in imbi.  i've noticed that when going to projects imbi is "loading" for about 5s.   it would be extremely useful to get this faster 

example URL where i see this issue:
https://$IMBI_FQDN/ui/projects/$PROJECT_ID

imbi.js is taking 557ms to fetch and blocks other ajax/async calls

cookie-cutters, environments, groups, namespaces, project-fact-types, project-link-types each take anywhere from 500-900ms to load on average but run mostly in parallel but theres quite a few and chrome limits the amount of concurrent downloads

this may not be as noticeable on imbi instances with a few projects but feels visually slower on instances with more then 100 projects

**Comment by pianoman19372** [comment]
![imbi](https://user-images.githubusercontent.com/643431/121766117-cd130d00-cb1d-11eb-883c-b1dfa1980845.png)

**Comment by gmr** [comment]
This is inherently due to the way the app is constructed. I've made some effort to improve the load time for the next version, but we have to go collect config data from the server on startup.

One way we could get around this is to cache the settings and metadata in local storage and use that, kicking off a refresh  in the background.

**Comment by gmr** [comment]
Closing as no longer relevant: This performance issue was caused by the old bundling approach (large imbi.js) and sequential data fetching in the v1 UI. Imbi 2.0 uses Vite for bundling and FastAPI for the backend, which resolves the underlying architecture issues described here.
